storage-operator/deploy/crds/storage_v1alpha1_volume_crd.yaml

```diff
@@ -67,6 +67,56 @@ spec:
           - storageClassName
           type: object
         status:
+          properties:
+            failedContext:
+              properties:
+                errorCode:
+                  enum:
+                  - 0
+                  - 1
+                  - 2
+                  - 3
+                  format: int64
+                  type: integer
+                errorMessage:
+                  type: string
+              type: object
+            pendingContext:
+              properties:
+                jobId:
+                  description: JID of the Salt job in progress.
+                  type: string
+                size:
+                  description: Size of the volume.
+                  type: string
+                step:
+                  description: Name of the current step.
+                  enum:
+                  - PreparationStarted
+                  - PrepareVolume
+                  - GetVolumeSize
+                  - CreatePersistentVolume
+                  type: string
+              type: object
+            phase:
+              enum:
+              - Available
+              - Pending
+              - Failed
+              - Terminating
+              type: string
+            terminatingContext:
+              properties:
+                jobId:
+                  type: string
+                step:
```

Comment:
Same as above. I'm afraid this will not work if Salt no longer 'knows' the JobID we store above, and such.

Reply:
When polling an unkown job ID you get something like:

```json
{
  "info": [
    {
      "jid": "20190725225251675732",
      "Result": {},
      "StartTime": "2019, Jul 25 22:52:51.675732",
      "Error": "Cannot contact returner or no job with this jid"
    }
  ],
  "return": [
    {}
  ]
}
```

I handle this case in my Salt client, and in the reconciliation loop I clear the JobId in such case (that way we will respawn a new Salt job).